docs/rfd.md

```diff
@@ -0,0 +1,478 @@
+  // This will first attempt to stop the job's process using SIGTERM, but if the
+  // process does not exit within a short grace period, it will be forcefully
+  // killed with SIGKILL.
+  //
+  // A job must be in the Running state to be stopped. If the job is in any
+  // other state, this returns a FailedPrecondition error.
+  rpc Stop(JobId) returns (google.protobuf.Empty);
+
+  // Returns the status of an existing job.
+  //
+  // If the job is completed, the exit code and detailed wait status will be
+  // present in the response. Jobs stopped by the user with the Stop() method
+  // may not have an exit code, depending if the process was able to exit
+  // cleanly within the grace period. Jobs that failed to start, or were
+  // otherwise killed by the system, will not have an exit code, but may have
+  // a detailed wait status indicating the reason for failure.
+  rpc Status(JobId) returns (JobStatus);
+
+  // Returns a list of all job IDs that are currently known to the server.
+  //
+  // All jobs, regardless of state, are included. No guarantees are made about
+  // the order of ids in the list.
+  rpc List(google.protobuf.Empty) returns (JobIdList);
+
+  // Streams the combined stdout and stderr output of a running or completed job.
+  // The output always starts from the beginning of process execution. Jobs that
+  // were stopped by the user with the Stop() method will have output up to the
+  // time they were stopped.
+  //
+  // If 'follow' is set to true, the output will continue to be be streamed in
+  // real-time until the job completes. Otherwise, only the currently accumulated
+  // output will be written, and the stream will be closed even if the job is
+  // still running.
+  //
+  // If the job is already completed, the full output of the job will be
+  // written to the stream, after which the stream will be closed.
+  rpc Output(OutputRequest) returns (stream ProcessOutput);
+}
+
+// JobSpec describes a command to be run, along with optional resource limits
+// that should be applied to the command's process.
+message JobSpec {
+  CommandSpec    command = 1;
+  ResourceLimits limits  = 2;
+}
+
+message JobId {
+  string id = 1;
+}
+
+message JobIdList {
+  repeated JobId items = 1;
+}
+
+enum State {
+  Unknown = 0;
+  // The job is waiting to be started, and is not yet running.
+  Pending = 1;
+  // The job is currently running.
+  Running = 2;
+  // The job exited normally (with any exit code), or was terminated via signal.
+  Completed = 3;
+  // The job failed to start, or exited abnormally due to an I/O error.
+  Failed = 5;
+}
+
+message JobStatus {
+  // The job's logical state.
+  State state = 1;
+  // A human-readable message describing the job's state, or an error message
+  // if the job failed.
+  string message = 2;
+  // The PID of the job's process.
+  int32 pid = 3;
+  // The exit code of the job's process.
+  // Only present if the job is in the Completed state.
+  optional int32 exitCode = 4;
```

Comment:
nit: the raw exit code is technically platform-dependent, so it might be worth splitting into actual exit code and signal exit instead